src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -312,8 +397,16 @@ ShenandoahCardCluster<RememberedSet>::block_start(const size_t card_index) const
   size_t offset = get_last_start(cur_index);
   // can avoid call via card size arithmetic below instead
   p = _rs->addr_for_card_index(cur_index) + offset;
-  // Recall that we already dealt with the co-initial object case above
+  // In the case that no object has been allocated in this card, we may not have dealt with co-initial object case above.
```

Comment:
See comment above. I suppose we should examine the allocation paths to see when objects are allocated in cards and when cards dirtied.